plugin/vim-jdb.vim

```diff
@@ -72,36 +77,54 @@ endfunction
 
 function! JdbOutHandler(channel, msg)
   "TODO make debug logging out of it
-  "echom a:msg
+  echom a:msg
   let l:breakpoint = ''
-  if -1 < stridx(a:msg, 'Breakpoint hit:')
+  if -1 < stridx(a:msg, 'Breakpoint hit: "thread')
     echom "breakpoint hit"
     let l:breakpoint = split(a:msg, ',')
     let l:filename = l:breakpoint[1]
-    let l:filename = substitute(l:filename, '\.\a*()$', '', '')
+    let l:filename = substitute(l:filename, '\.\w*()$', '', '')
     let l:filename = substitute(l:filename, ' ', '', 'g')
-    let l:filename = join(split(l:filename, '\.'), '/')
+    let l:filenamefrag= split(l:filename, '\.')
+    let l:filename = findfile(filenamefrag[-1])
+    echom 'found file ' . l:filename
     let l:linenumber = substitute(l:breakpoint[2], ',\|\.\| \|bci=\d*\|line=', '', 'g')
-    " TODO only open when current buffer is not the file to open
-    exe 'e +%foldopen! **/'. l:filename .'.java'
-    exe l:linenumber
-    exe 'sign unplace 2'
-    exe 'sign place 2 line='. l:linenumber .' name=currentline file='.  expand("%:p")
+    " only open when current buffer is not the file to open
+    if l:filename != s:currentfile
+      if filereadable(l:filename)
+        exe 'e +%foldopen! **/'. l:filename 
+	let s:currentfile = l:filename
+        exe l:linenumber
+        exe 'sign unplace 2'
+        exe 'sign place 2 line='. l:linenumber .' name=currentline file='.  expand("%:p")
+      else 
+	echom 'tried to open file ' . l:filename
+      endif
+    endif
   endif
-  if -1 < stridx(a:msg, 'Step completed:')
+  if -1 < stridx(a:msg, 'Step completed: "thread')
     " TODO handle ClassName$3.get()
     echom "Step completed"
     let l:breakpoint = split(a:msg, ',')
     let l:filename = l:breakpoint[1]
-    let l:filename = substitute(l:filename, '\.\a*()$', '', '')
+    let l:filename = substitute(l:filename, '\.\w*()$', '', '')
     let l:filename = substitute(l:filename, ' ', '', 'g')
-    let l:filename = join(split(l:filename, '\.'), '/')
+    let l:filenamefrag= split(l:filename, '\.')
+    let l:filename = findfile(filenamefrag[-1])
+    echom 'found file ' . l:filename
     let l:linenumber = substitute(l:breakpoint[2], ',\|\.\| \|bci=\d*\|line=', '', 'g')
-    " TODO only open when current buffer is not the file to open
-    exe 'e +%foldopen! **/'. l:filename .'.java'
-    exe l:linenumber
-    exe 'sign unplace 2'
-    exe 'sign place 2 line='. l:linenumber .' name=currentline file='.  expand("%:p")
+    " only open when current buffer is not the file to open
+    " if l:filename != l:currentfile
+      if filereadable(l:filename)
+        exe 'e +%foldopen! **/'. l:filename
+	let s:currentfile = l:filename
+        exe l:linenumber
+        exe 'sign unplace 2'
+        exe 'sign place 2 line='. l:linenumber .' name=currentline file='.  expand("%:p")
```

Comment:
You no longer call this for the file open case. How is the sign placed in this case?